Sources/SmithyReadWrite/DocumentReadingClosure.swift

```diff
@@ -9,6 +9,14 @@ import struct Foundation.Data
 
 public typealias DocumentReadingClosure<T, Reader> = (Data, ReadingClosure<T, Reader>) throws -> T
```

Comment:
This type was moved in from `ClientRuntime`, and was made generic to any `SmithyReader`.